catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -564,47 +581,52 @@ private ContentItem generateContentFile(
 
     Path contentItemPath =
         Paths.get(contentDirectory.toAbsolutePath().toString(), item.getFilename());
-    ByteSource byteSource;
-
-    long copy;
+    byte[] plainBytes;
+    byte[] encryptedBytes;
 
     if (storeReference != null) {
-      copy = item.getSize();
-      Files.write(
-          Paths.get(contentItemPath.toString() + "." + REF_EXT),
-          storeReference.getBytes(Charset.forName("UTF-8")));
-      byteSource =
-          new ByteSource() {
-            @Override
-            public InputStream openStream() throws IOException {
-              return new URL(storeReference).openStream();
-            }
-          };
+      plainBytes = storeReference.getBytes();
+      encryptedBytes = encryptorDecryptor.encrypt(plainBytes);
```

Comment:
❓ What if the data is already stored on an encrypted partition or the data is not sensitive enough to require encryption but instead requires performance?  Should we make this configurable?

Reply:
We could have a config flag for this that disables encryption if you don't want it. Seems fine to me.

Reply:
I doubt this would really impact performance all that much. I think you'd run into a lot of other long poles before this one ended up being a real issue.

Reply:
✏️ Maybe change the interface to be more straight `ReadContent`/`WriteContent` interfaces and allow for multiple implementations to be deployed at a time. Then we could do a lookup for the correct reader/writer at runtime. Some would just pass through the stream and others would encrypt/decrypt as necessary.

Reply:
Great idea @coyotesqrl! The only problem is that that would require a larger design change, whereas this change is just throwing encryption on our file system interactions.

To further elaborate: trying to simply put encryption in the `FileSystemStorageProvider` has proven to be quite difficult. When passing all of these streams around, I need to making sure they are opening/closing when they need to be. I would be afraid that making the larger design change you recommended would take a much larger amount of time; that time spent on something that isn't necessary atm.